corehq/apps/motech/openmrs/concepts/sync.py

```diff
@@ -0,0 +1,112 @@
+import json
+import jsonobject
+from corehq.apps.motech.openmrs.concepts.models import OpenmrsConcept
+from corehq.apps.motech.openmrs.restclient.listapi import OpenmrsListApi
+
+
+def sync_concepts_from_openmrs(account):
+    requests = account.get_requests_object()
+    OpenmrsConcept.objects.filter(account=account).delete()
+    api = OpenmrsListApi(requests, 'concept')
+    answers_relationships = []
+    for concept in api.get_all():
+        concept = openmrs_concept_json_from_api_json(concept)
+        concept, answers = openmrs_concept_from_concept_json(account, concept)
+        if answers:
+            answers_relationships.append((concept, answers))
+        concept.save()
+
+    while answers_relationships:
```

Comment:
Concepts correspond to a few different things as we think of them in CommCare, including
- questions
- answers
- case properties (e.g. patient properties)
- case property values (e.g. patient property values)

A concept that is... _conceptually_... a "question", will be defined with "answers" which are other concepts.

The key lines in here is

```python
concept.answers = answer_concepts
concept.save()
```

Conceptually this is really just part of the the previous loop—for every concept that comes in you want to save the corresponding answers in the question & answer table, which is a many-to-many table from concepts (questions) to concepts (answers). But you can't save to that table until both the question and answer involved in a row has been saved. Hence, I defer populating that table until each concept has been saved.